LibWeb(+LibWebView+WebContent): Handle double clicks

krkk opened the pull request:
- **LibWeb: Add stub implementation for handling doubleclicks**
- **LibWeb+LibWebView+WebContent: Get doubleclick events from LibGUI**
- **LibWeb: Dispatch `dblclick` UI Events on double click**

  https://user-images.githubusercontent.com/16520278/173686406-f0fa1914-e2f2-420b-b757-3a7cafaff6d7.mp4

- **LibWeb: Make doubleclicking select the nearest word**

  https://user-images.githubusercontent.com/16520278/173686389-513d11dc-ab04-4b08-bd88-869eb05146de.mp4

AtkinsSJ commented on Userland/Libraries/LibWeb/Page/EventHandler.cpp:
```diff
@@ -545,7 +545,41 @@ bool EventHandler::handle_doubleclick(Gfx::IntPoint const& position, unsigned bu
+
+            int first_word_break_before = [&] {
+                for (size_t i = result->index_in_node - 1; i > 0; --i) {
+                    if (text_for_rendering[i] == ' ')
```
This should probably check if the character is whitespace instead. For example, double-clicking on "mousemove" in the mouse-events test page highlighted this, because it doesn't treat the newline as whitespace:
![image](https://user-images.githubusercontent.com/222642/174083303-5f8ddbcf-7155-4999-9df8-66e04f754bf4.png)

AtkinsSJ commented on Userland/Libraries/LibWeb/Page/EventHandler.cpp:
```diff
@@ -545,7 +545,41 @@ bool EventHandler::handle_doubleclick(Gfx::IntPoint const& position, unsigned bu
+
+            int first_word_break_after = [&] {
+                for (size_t i = result->index_in_node; i < text_for_rendering.length(); ++i) {
+                    if (text_for_rendering[i] == ' ')
```
Same here.

AtkinsSJ reviewed (changes requested):
Nice. :^)

The double-click text selection feels a little awkward since it only selects after you release the mouse button, but it's still better than not having it.

krkk commented:
Changes since last push:
- added some comments about not so obvious `- 1` and `+ 1` in the first loop
- made first loop variable signed, because subtracting the index when it was 0 would overflow and crash the page (could be seen on google)
- run the loop also for the first character (`i >= 0`) – previously selecting text in `<pre> text</pre>` also selected the space.
